v2/emailpassword/common-customizations/signup-form/adding-custom-components.mdx

```diff
@@ -0,0 +1,334 @@
+                                <span style={{ marginLeft: 5 }}>I agree to the terms and conditions</span>
+                            </div>
+                        )
+                    }]
+                }
+                // highlight-end
+            }
+        }),
+        Session.init()
+    ]
+});
+```
+</TabItem>
+</FrontendSDKTabs>
+
+<img alt="Prebuilt form UI with custom components" src="/img/emailpassword/signup-with-custom-components.png" />
+
+## Step 2: Back End
+
+### Add fields to SuperTokens `init`
+Now that you have added new fields to the front end you need to make sure that the backend will take them into account when your new users register.
+
+Go back to where you have called the SuperTokens init function in your backend application:
+
+<BackendSDKTabs>
+<TabItem value="nodejs">
+
+```tsx
+import SuperTokens from "supertokens-node";
+import EmailPassword from "supertokens-node/recipe/emailpassword";
+import Session from "supertokens-node/recipe/session";
+
+SuperTokens.init({
+  appInfo: {
+    apiDomain: "...",
+    appName: "...",
+    websiteDomain: "..."
+  },
+  supertokens: {
+    connectionURI: "...",
+  },
+  recipeList: [
+    EmailPassword.init({
+      signUpFeature: {
+        // highlight-start
+        formFields: [{
+          id: "select-dropdown"
+        }, {
+          id: "terms"
+        }]
+        // highlight-end
+      }
+    }),
+    Session.init({ /* ... */ })
+  ]
+});
+```
+
+</TabItem>
+<TabItem value="go">
+
+```go
+import (
+	"github.com/supertokens/supertokens-golang/recipe/emailpassword"
+	"github.com/supertokens/supertokens-golang/recipe/emailpassword/epmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+func main() {
+	countryOptional := true
+	supertokens.Init(supertokens.TypeInput{
+		RecipeList: []supertokens.Recipe{
+			emailpassword.Init(&epmodels.TypeInput{
+				SignUpFeature: &epmodels.TypeInputSignUp{
+					//highlight-start
+					FormFields: []epmodels.TypeInputFormField{
+						{
+							ID: "select-dropdown",
+						},
+						{
+							ID: "terms",
+						},
+					},
+					//highlight-end
+				},
+			}),
+		},
+	})
+}
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python
+from supertokens_python import init, InputAppInfo
+from supertokens_python.recipe import emailpassword, session
+from supertokens_python.recipe.emailpassword import InputFormField
+
+init(
+    app_info=InputAppInfo(api_domain="...", app_name="...", website_domain="..."),
+    
+    framework='...', # type: ignore
+    recipe_list=[
+        emailpassword.init(
+            # highlight-start
+            sign_up_feature=emailpassword.InputSignUpFeature(
+                form_fields=[InputFormField(id='select-dropdown'), InputFormField(id='terms')]
+            )
+            # highlight-end
+        ),
+        session.init()
+    ]
+)
+```
+
+</TabItem>
+</BackendSDKTabs>
+
+## Step 3: Handle form fields on successful Sign-up
```

Comment:
you don't need this.. cause it's already there in https://supertokens.com/docs/emailpassword/common-customizations/signup-form/adding-fields